presto-main/src/main/java/com/facebook/presto/sql/relational/Expressions.java

```diff
@@ -113,6 +130,39 @@ public static CallExpression call(FunctionAndTypeManager functionAndTypeManager,
         return call(name, functionHandle, returnType, arguments);
     }
 
+    public static RowExpression searchedCaseExpression(List<RowExpression> whenClauses, Optional<RowExpression> defaultValue)
+    {
+        // We rewrite this as - CASE true WHEN p1 THEN v1 WHEN p2 THEN v2 .. ELSE v END
+        return buildSwitch(new ConstantExpression(true, BOOLEAN), whenClauses, defaultValue, BOOLEAN);
+    }
+
+    public static RowExpression buildSwitch(RowExpression operand, List<RowExpression> whenClauses, Optional<RowExpression> defaultValue, Type returnType)
```

Comment:
move to PlannerUtils?

Reply:
This class has a bunch of utils already to create expressions, so just using that.